### ISSUE TITLE
Install "from source" fails on node LTS

### ISSUE THREAD

**Issue by sthagen** [secvisogram/secvisogram#214]
As advertized in #213 the app (secvisogram `sha1:d7843aa9`) is not installable anymore (on my machine) when using the LTS node (and bundled npm) version:
```console
❯ for app in node npm; do printf "%s --> %s\n" "$app" "$("$app" --version)"; done
node --> v16.13.1
npm --> 8.1.2
```

Given a clean clone and having executed the recursive git submodule updates.
- When executing `npm ci` (with `elsewhere`--> kind of `$HOME`, `somewhere` --> parent folder of repository):
  - Then:
	```console
	❯ npm ci
	npm WARN old lockfile
	npm WARN old lockfile The package-lock.json file was created with an old version of npm,
	npm WARN old lockfile so supplemental metadata must be fetched from the registry.
	npm WARN old lockfile
	npm WARN old lockfile This is a one-time fix-up, please be patient...
	npm WARN old lockfile

	> postinstall
	> npm run -s install:app && npm run -s install:bin

	npm ERR! code 1
	npm ERR! path /somewhere/secvisogram-secvisogram
	npm ERR! command failed
	npm ERR! command sh -c npm run -s install:app && npm run -s install:bin

	npm ERR! A complete log of this run can be found in:
	npm ERR!     /elsewhere/.npm/_logs/2021-12-16T13_02_56_413Z-debug.log
	# return code 1
	```

The aforementioned log:
```console
❯ cat /elsewhere/.npm/_logs/2021-12-16T13_02_56_413Z-debug.log
0 verbose cli [
0 verbose cli   '/elsewhere/.nvm/versions/node/v16.13.1/bin/node',
0 verbose cli   '/elsewhere/.nvm/versions/node/v16.13.1/bin/npm',
0 verbose cli   'ci'
0 verbose cli ]
1 info using npm@8.1.2
2 info using node@v16.13.1
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 3ms
5 timing config:load:file:/elsewhere/.nvm/versions/node/v16.13.1/lib/node_modules/npm/npmrc Completed in 2ms
6 timing config:load:builtin Completed in 2ms
7 timing config:load:cli Completed in 3ms
8 timing config:load:env Completed in 1ms
9 timing config:load:file:/somewhere/secvisogram-secvisogram/.npmrc Completed in 0ms
10 timing config:load:project Completed in 1ms
11 timing config:load:file:/elsewhere/.npmrc Completed in 3ms
12 timing config:load:user Completed in 3ms
13 timing config:load:file:/elsewhere/.nvm/versions/node/v16.13.1/etc/npmrc Completed in 0ms
14 timing config:load:global Completed in 1ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:credentials Completed in 2ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 17ms
19 timing npm:load:configload Completed in 17ms
20 timing npm:load:setTitle Completed in 13ms
21 timing npm:load:setupLog Completed in 1ms
22 timing config:load:flatten Completed in 2ms
23 timing npm:load:cleanupLog Completed in 2ms
24 timing npm:load:configScope Completed in 0ms
25 timing npm:load:projectScope Completed in 2ms
26 timing npm:load Completed in 39ms
27 timing arborist:ctor Completed in 6ms
28 timing npm-ci:rm Completed in 8ms
29 timing idealTree:init Completed in 1ms
30 warn old lockfile The package-lock.json file was created with an old version of npm,
30 warn old lockfile so supplemental metadata must be fetched from the registry.
30 warn old lockfile
30 warn old lockfile This is a one-time fix-up, please be patient...
31 silly inflate node_modules/prettier
32 silly inflate node_modules/typescript
33 http fetch GET 200 https://registry.npmjs.org/prettier 241ms (cache revalidated)
34 http fetch GET 200 https://registry.npmjs.org/typescript 588ms (cache updated)
35 timing idealTree:inflate Completed in 623ms
36 timing idealTree:userRequests Completed in 0ms
37 silly idealTree buildDeps
38 timing idealTree:#root Completed in 0ms
39 timing idealTree:buildDeps Completed in 1ms
40 timing idealTree:fixDepFlags Completed in 0ms
41 timing idealTree Completed in 625ms
42 timing reify:loadTrees Completed in 626ms
43 timing reify:diffTrees Completed in 0ms
44 silly reify moves {}
45 timing reify:retireShallow Completed in 0ms
46 timing reify:createSparse Completed in 2ms
47 timing reify:loadBundles Completed in 0ms
48 silly audit bulk request { prettier: [ '2.3.2' ], typescript: [ '4.4.4' ] }
49 http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 346ms
50 timing auditReport:getReport Completed in 346ms
51 silly audit report {}
52 timing auditReport:init Completed in 0ms
53 timing reify:audit Completed in 346ms
54 timing reifyNode:node_modules/prettier Completed in 381ms
55 timing reifyNode:node_modules/typescript Completed in 462ms
56 timing reify:unpack Completed in 462ms
57 timing reify:unretire Completed in 0ms
58 timing build:queue Completed in 1ms
59 timing build:link:node_modules/prettier Completed in 2ms
60 timing build:link:node_modules/typescript Completed in 2ms
61 timing build:link Completed in 2ms
62 timing build:deps Completed in 3ms
63 timing build Completed in 3ms
64 timing reify:build Completed in 3ms
65 timing reify:trash Completed in 0ms
66 timing reify Completed in 1099ms
67 timing command:ci Completed in 18284ms
68 verbose stack Error: command failed
68 verbose stack     at ChildProcess.<anonymous> (/elsewhere/.nvm/versions/node/v16.13.1/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
68 verbose stack     at ChildProcess.emit (node:events:390:28)
68 verbose stack     at maybeClose (node:internal/child_process:1064:16)
68 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
69 verbose cwd /somewhere/secvisogram-secvisogram
70 verbose Darwin 20.6.0
71 verbose argv "/elsewhere/.nvm/versions/node/v16.13.1/bin/node" "/elsewhere/.nvm/versions/node/v16.13.1/bin/npm" "ci"
72 verbose node v16.13.1
73 verbose npm  v8.1.2
74 error code 1
75 error path /somewhere/secvisogram-secvisogram
76 error command failed
77 error command sh -c npm run -s install:app && npm run -s install:bin
78 verbose exit 1
```

**Comment by mfd2007** [comment]
There is maybe a problem with the download of FontAwesome. 

When running `npm run -s install:app` directly an not via `npm ci` there a some hints in the log. 

```
2914 verbose stack HttpErrorAuthUnknown: Unable to authenticate, need: Basic realm="https://npm.fontawesome.com/",service="npm.fontawesome.com"
2914 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/npm-registry-fetch/check-response.js:84:17
2914 verbose stack     at runMicrotasks (<anonymous>)
2914 verbose stack     at processTicksAndRejections (node:internal/process/task_queues:96:5)
2915 verbose statusCode 401
2916 verbose pkgid @fortawesome/free-solid-svg-icons@https://npm.fontawesome.com/@fortawesome/free-solid-svg-icons/-/5.15.4/free-solid-svg-icons-5.15.4.tgz
```

**Comment by mfd2007** [comment]
After installing the following packages manually, I can compilation and execution is successful. 

```
npm i @fortawesome/fontawesome-free@5.15.4
npm i @fortawesome/free-solid-svg-icons
npm i @fortawesome/fontawesome-svg-core
```

But I don't know how to fix the automated installation.

**Comment by sthagen** [comment]
@mfd2007 I think you mean (`@fortawesome`--> `@fontawesome`:

```
npm i @fontawesome/fontawesome-free@5.15.4
npm i @fontawesome/free-solid-svg-icons
npm i @fontawesome/fontawesome-svg-core
```

> But I don't know how to fix the automated installation.

Maybe add this to some package.json ?

**Comment by tschmidtb51** [comment]
I had that nasty bug once before. I just checked: The problem is that there is a non-free version of the fortawesome stuff. When the `app/package-lock.json` lists `https://npm.fontawesome.com/` instead of `https://registry.npmjs.org` we run in that problem.

We need to do two things here:
1. Fix the problem.
2. Prevent that this can happen again.

For the second part, we should add a GitHub action which will check whether the `app/package-lock.json`contains `https://npm.fontawesome.com/`. If so the action shall fail.

Adding @domachine, @wrtlbrmpft for attention.

**Comment by sthagen** [comment]
Ah, good catch @tschmidtb51 - ok, I personally support / pay for font awesome, so maybe my machine is not the best reference (anyway).

**Comment by domachine** [comment]
I'm sorry for messing this up. Could you check if this fixes the issue for you? https://github.com/secvisogram/secvisogram/pull/220

**Comment by sthagen** [comment]
> I'm sorry for messing this up. Could you check if this fixes the issue for you? #220

Someone has to make changes - no problem for me; cf. my comment at [#220](https://github.com/secvisogram/secvisogram/pull/220#pullrequestreview-835434415) will test this evening.
